components/library/src/content/mod.rs

```diff
@@ -28,6 +29,12 @@ pub fn has_anchor(headings: &[Heading], anchor: &str) -> bool {
     false
 }
 
+
+pub fn has_anchor_id(content: &str, anchor: &str) -> bool {
```

Comment:
I would probably move this function to utils directly since we will only ever do `is_match`